src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -72,6 +72,24 @@ internal sealed class LocalFunctionSymbol : MethodSymbol
                 ReportAttributesDisallowed(param.AttributeLists, _declarationDiagnostics);
             }
 
+            if (syntax.ReturnType.Kind() == SyntaxKind.RefType)
+            {
+                var returnType = (RefTypeSyntax)syntax.ReturnType;
+                if (returnType.ReadOnlyKeyword.Kind() == SyntaxKind.ReadOnlyKeyword)
+                {
+                    _refKind = RefKind.RefReadOnly;
+                    DeclaringCompilation.EnsureIsReadOnlyAttributeExists(_declarationDiagnostics, returnType.Location, modifyCompilation: false);
```

Comment:
>DeclaringCompilation.EnsureIsReadOnlyAttributeExists(_declarationDiagnostics, returnType.Location, modifyCompilation: false); [](start = 20, length = 125)

We should not be doing this in a constructor. Creation of a symbol shouldn't trigger binding or symbol lookup. ComputeReturnType looks like a good place to do this.